codegen/builtin_fs.go

```diff
@@ -31,6 +32,20 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+const (
+	// HistoryComment is an indicator in the image history that it the history
+	// layer was produced by the HLB compiler.
+	HistoryComment = "hlb.v0"
+)
+
+func commitHistory(img *specs.Image, format string, a ...interface{}) {
+	img.History = append(img.History, specs.History{
+		CreatedBy:  fmt.Sprintf(format, a...),
+		Comment:    HistoryComment,
+		EmptyLayer: true,
```

Comment:
Is it correct for `EmptyLayer` to always be true? Is the implication that the downstream tools don't really pay attention to it? (Still, wondering if `false` would be safer.)

Reply:
EmptyLayer should be `false` for `RUN`, `COPY`, `ADD`, but `true` for everything else. Let me see if I can add history for `copy` as well.